drivers/sensor/bmi08x/bmi08x_gyro.c

```diff
@@ -0,0 +1,472 @@
+	return i2c_write_dt(&bmi08x->bus.i2c, buf, 1 + length);
+}
+
+static int bmi08x_bus_check_i2c(const union bmi08x_bus *bus)
+{
+	return device_is_ready(bus->i2c.bus) ? 0 : -ENODEV;
+}
+
+static const struct bmi08x_gyro_bus_io bmi08x_i2c_api = {
+	.check = bmi08x_bus_check_i2c,
+	.transceive = bmi08x_gyro_transceive_i2c,
+};
+
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(i2c) */
+
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+
+static int bmi08x_gyro_transceive_spi(const struct device *dev, uint8_t reg, bool write, void *data,
+				      size_t length)
+{
+	const struct bmi08x_gyro_config *bmi08x = dev->config;
+	const struct spi_buf tx_buf[2] = {{.buf = &reg, .len = 1}, {.buf = data, .len = length}};
+	const struct spi_buf_set tx = {.buffers = tx_buf, .count = write ? 2 : 1};
+
+	if (!write) {
+		uint16_t dummy;
+		const struct spi_buf rx_buf[2] = {{.buf = &dummy, .len = 1},
+						  {.buf = data, .len = length}};
+		const struct spi_buf_set rx = {.buffers = rx_buf, .count = 2};
+
+		return spi_transceive_dt(&bmi08x->bus.spi, &tx, &rx);
+	}
+
+	return spi_write_dt(&bmi08x->bus.spi, &tx);
+}
+
+static int bmi08x_bus_check_spi(const union bmi08x_bus *bus)
+{
+	return spi_is_ready_dt(&bus->spi) ? 0 : -ENODEV;
+}
+
+static const struct bmi08x_gyro_bus_io bmi08x_spi_api = {
+	.check = bmi08x_bus_check_spi,
+	.transceive = bmi08x_gyro_transceive_spi,
+};
+
+#endif /* DT_ANY_INST_ON_BUS_STATUS_OKAY(spi) */
+
+static inline int bmi08x_bus_check(const struct device *dev)
+{
+	const struct bmi08x_gyro_config *config = dev->config;
+
+	return config->api->check(&config->bus);
+}
+
+static int bmi08x_gyro_transceive(const struct device *dev, uint8_t reg, bool write, void *data,
+				  size_t length)
+{
+	const struct bmi08x_gyro_config *cfg = dev->config;
+
+	return cfg->api->transceive(dev, reg, write, data, length);
+}
+
+int bmi08x_gyro_read(const struct device *dev, uint8_t reg_addr, uint8_t *data, uint8_t len)
+{
+	return bmi08x_gyro_transceive(dev, reg_addr | BIT(7), false, data, len);
+}
+
+int bmi08x_gyro_byte_read(const struct device *dev, uint8_t reg_addr, uint8_t *byte)
+{
+	return bmi08x_gyro_transceive(dev, reg_addr | BIT(7), false, byte, 1);
+}
+
+int bmi08x_gyro_byte_write(const struct device *dev, uint8_t reg_addr, uint8_t byte)
+{
+	return bmi08x_gyro_transceive(dev, reg_addr & 0x7F, true, &byte, 1);
+}
+
+int bmi08x_gyro_word_write(const struct device *dev, uint8_t reg_addr, uint16_t word)
+{
+	uint8_t tx_word[2] = {(uint8_t)(word & 0xff), (uint8_t)(word >> 8)};
+
+	return bmi08x_gyro_transceive(dev, reg_addr & 0x7F, true, tx_word, 2);
+}
+
+int bmi08x_gyro_reg_field_update(const struct device *dev, uint8_t reg_addr, uint8_t pos,
+				 uint8_t mask, uint8_t val)
+{
+	uint8_t old_val;
+
+	if (bmi08x_gyro_byte_read(dev, reg_addr, &old_val) < 0) {
+		return -EIO;
+	}
+
+	return bmi08x_gyro_byte_write(dev, reg_addr, (old_val & ~mask) | ((val << pos) & mask));
+}
+
+static const struct bmi08x_range bmi08x_gyr_range_map[] = {
```

Comment:
The sorting here should be from small to large

Reply:
No, the reg value goes from large to small, please take look at reg 0x0F: GYRO_RANGE in the datasheet